packages/alfa-rules/src/sia-er62/rule.ts

```diff
@@ -135,20 +143,69 @@ export default Rule.Atomic.of<Page, Element>({
 
         const hasDistinguishingStyle = (context: Context = Context.empty()) =>
           Set.from(
-            linkElements.map((link) =>
+            linkElements.map((link) => {
               // If the link element is distinguishable from at least one
               // non-link element, this is good enough.
               // Note that ACT rules draft requires the link-element to be
               // distinguishable from *all* non-link elements in order to be good.
-              nonLinkElements.some(
+              const hasDistinguishableStyle = nonLinkElements.some(
                 (container) =>
-                  isDistinguishable(container, device, context)(link) ||
+                  Distinguishable.isDistinguishable(
+                    container,
+                    device,
+                    context
+                  )(link) ||
                   (context.isHovered(target) &&
-                    hasDistinguishableCursor(container, device, context)(link))
-              )
-                ? Ok.of(ComputedStyles.from(link, device, target, context))
-                : Err.of(ComputedStyles.from(link, device, target, context))
-            )
+                    Distinguishable.hasDistinguishableCursor(
+                      container,
+                      device,
+                      context
+                    )(link))
+              );
+
+              const distinguishableContrast = Set.from(
+                nonLinkElements.flatMap((container) =>
+                  Sequence.from(
+                    pairwiseContrastCache
+                      .get(device, Cache.empty)
+                      .get(context, Cache.empty)
+                      .get(container, Cache.empty)
+                      .get(link, () =>
+                        Distinguishable.getPairwiseContrast(
+                          container,
+                          link,
+                          device,
+                          context
+                        )
+                      )
+                  )
+                )
+              ).toArray();
+
+              return hasDistinguishableStyle
+                ? Ok.of(
+                    ElementDistinguishable.of(
+                      ElementDistinguishable.serialise(
+                        link,
+                        device,
+                        target,
+                        context
+                      ),
+                      distinguishableContrast
+                    )
```

Comment:
Let's make an `ElementDistinguishable.from(link, device, target, context, pairings)` that takes care of calling `.serialise`.
Again, no need for the caller to know it has to do some serialisation, and future versions of this diagnostic could well handle it differently.


As a rule of thumb, we often have:
* In the class, a private constructor that takes the data in the same shape as the the `private readonly` variables storing it (and as flat as needed, the getters may nest data that makes sense being nested).
* In the class, a `public static of` that takes the data almost in the same shape as the constructor, maybe taking care of turning any `Iterable` into the correct kind, or having optional arguments turned into actual `Option`.
* In the namespace, if needed, a `.from` that takes data in a fairly different format and perform some possibly complex computation to turn it into the correct format.

Here, `Element.serialise` is the old `ComputedStyles.from`, but it is more in line with the rest of the code to keep that one flat also; even if the resulting `.from` might end up calling subfunction to modify each piece of data in a different way.